.github/workflows/aws-e2e.yaml

```diff
@@ -0,0 +1,105 @@
+'on':
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - '*'
+permissions:
+  contents: read
+  id-token: write
+env:
+  PACKER_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+name: AWS
+jobs:
```

Comment:
We also need to invoke these jobs on-demand from GHA UI using selecting "branch name", "build config" and "OS".
What do you think of using reusable-workflows. 
you can create another workflow with `workflow_call` and `workflow_dispatch` and call that work from current workflow which executes on PR. 

we did this in konvoy2.
caller workflow: https://github.com/mesosphere/konvoy2/blob/main/.github/workflows/daily_e2e_tests.yaml#L50-L54
which calls:
https://github.com/mesosphere/konvoy2/blob/main/.github/workflows/aws_e2e_tests.yaml#L10-L42

Reply:
btw, if we decide to use reusable workflows, we can add this enhancement in follow up PR. This PR does not need to get complicated now.